libsolidity/analysis/DocStringAnalyser.cpp

```diff
@@ -32,6 +32,52 @@ using namespace solidity;
 using namespace solidity::langutil;
 using namespace solidity::frontend;
 
+namespace
+{
+
+void copyMissingTags(StructurallyDocumentedAnnotation& _target, CallableDeclarationAnnotation const& _source)
+{
+	auto& sourceDoc = dynamic_cast<StructurallyDocumentedAnnotation const&>(_source);
+
+	set<string> existingTags;
+
+	for (auto& iterator: _target.docTags)
+		existingTags.insert(iterator.first);
+
+	for (auto& [tag, content]: sourceDoc.docTags)
+		if (tag != "inheritdoc" && !existingTags.count(tag))
+			_target.docTags.emplace(tag, content);
+}
+
+CallableDeclaration const* findBaseCallable(std::set<CallableDeclaration const*>& _baseFuncs, string const& _name)
+{
+	for (CallableDeclaration const* baseFuncCandidate: _baseFuncs)
+		if (baseFuncCandidate->annotation().contract->name() == _name)
+			return baseFuncCandidate;
+		else if (auto callable = findBaseCallable(baseFuncCandidate->annotation().baseFunctions, _name))
+			return callable;
+
+	return nullptr;
```

Comment:
I wonder if it's possible to pick the contract first from  `ContractAnnotation::linearizedBaseContracts`, followed by the function? What do you think?

Reply:
What advantages do you hope to gain from that? If we use the contracts as basis (instead of the already deduced baseFunctions in the annotation), we still have to match the function name & parameters of each contract with this one, thus redoing work that was already done to get the content of `baseFunctions`..

Reply:
A minor advantage is that we can distinguish between "contract doesn't exist" and "contract doesn't contain a function that we want." But I agree that matching the function after finding the contract might be tricky.